pg_search/src/schema/config.rs

```diff
@@ -27,6 +27,7 @@ pub struct SearchConfig {
     pub index_name: String,
     pub index_oid: u32,
     pub table_oid: u32,
+    pub database_oid: u32,
```

Comment:
Why add this to `SearchConfig` when it's always available to us an otherwise an immutable property of a database?

Reply:
In non-connection processes, like the background writer process, `pg_sys::MyDatabaseId` doesn't work at all. It constantly returns `0`.

We don't have an _immediate_ use case for `SearchConfig` from the background writer.... but because there doesn't seem to be a reliable way to ask for `database_oid`, it doesn't seem like a bad idea to have a way for it to be retrieved from a serialized format.

Reply:
hmm.  Okay.  

I feel like as we move away from the background worker and towards concurrent writers SearchConfig is going to become much less complex, so for now I suppose this is fine.